packages/mdc-menu-surface/foundation.ts

```diff
@@ -372,61 +400,64 @@ class MDCMenuSurfaceFoundation extends MDCFoundation {
    */
-  getMenuSurfaceMaxHeight_(corner) {
+  private getMenuSurfaceMaxHeight_(corner: Corner): number {
     let maxHeight = 0;
-    const {viewportDistance} = this.measures_;
+    const {viewportDistance} = this.measurements_;
+
     const isBottomAligned = Boolean(corner & CornerBit.BOTTOM);
     const {MARGIN_TO_EDGE} = MDCMenuSurfaceFoundation.numbers;
 
     // When maximum height is not specified, it is handled from css.
     if (isBottomAligned) {
       maxHeight = viewportDistance.top + this.anchorMargin_.top - MARGIN_TO_EDGE;
       if (!(this.anchorCorner_ & CornerBit.BOTTOM)) {
-        maxHeight += this.measures_.anchorHeight;
+        maxHeight += this.measurements_.anchorHeight;
       }
     } else {
-      maxHeight = viewportDistance.bottom - this.anchorMargin_.bottom + this.measures_.anchorHeight - MARGIN_TO_EDGE;
+      maxHeight =
+          viewportDistance.bottom - this.anchorMargin_.bottom + this.measurements_.anchorHeight - MARGIN_TO_EDGE;
       if (this.anchorCorner_ & CornerBit.BOTTOM) {
-        maxHeight -= this.measures_.anchorHeight;
+        maxHeight -= this.measurements_.anchorHeight;
       }
     }
 
     return maxHeight;
   }
 
-  /** @private */
-  autoPosition_() {
+  private autoPosition_() {
     // Compute measurements for autoposition methods reuse.
-    this.measures_ = this.getAutoLayoutMeasurements_();
+    this.measurements_ = this.getAutoLayoutMeasurements_();
 
     const corner = this.getOriginCorner_();
     const maxMenuSurfaceHeight = this.getMenuSurfaceMaxHeight_(corner);
     const verticalAlignment = (corner & CornerBit.BOTTOM) ? 'bottom' : 'top';
     let horizontalAlignment = (corner & CornerBit.RIGHT) ? 'right' : 'left';
     const horizontalOffset = this.getHorizontalOriginOffset_(corner);
     const verticalOffset = this.getVerticalOriginOffset_(corner);
-    let position = {
+    const {anchorWidth, surfaceWidth} = this.measurements_;
+
+    const position: MDCMenuSurfacePositionInput = {
       [horizontalAlignment]: horizontalOffset ? horizontalOffset : '0',
       [verticalAlignment]: verticalOffset ? verticalOffset : '0',
     };
-    const {anchorWidth, surfaceWidth} = this.measures_;
+
     // Center align when anchor width is comparable or greater than menu surface, otherwise keep corner.
     if (anchorWidth / surfaceWidth > numbers.ANCHOR_TO_MENU_SURFACE_WIDTH_RATIO) {
       horizontalAlignment = 'center';
     }
 
     // If the menu-surface has been hoisted to the body, it's no longer relative to the anchor element
-    if (this.hoistedElement_ || this.isFixedPosition_) {
-      position = this.adjustPositionForHoistedElement_(position);
+    if (this.isHoistedElement_ || this.isFixedPosition_) {
+      this.adjustPositionForHoistedElement_(position);
     }
 
     for (const prop in position) {
       if (position.hasOwnProperty(prop) && position[prop] !== '0') {
-        position[prop] = `${parseInt(position[prop], 10)}px`;
+        // tslint:disable-next-line:ban
```

Comment:
`parseInt()`. Luckily, I was able to remove all the `parseInt()` calls by modifying several private methods to pass `number` values to each other instead of `string`s with `"px"` at the end.